jib-core/src/main/java/com/google/cloud/tools/jib/cache/LayerEntriesSelector.java

```diff
@@ -83,10 +87,15 @@ public int compareTo(LayerEntryTemplate otherLayerEntryTemplate) {
       if (extractionPathComparison != 0) {
         return extractionPathComparison;
       }
-      int lastModifiedTimeComparison =
-          lastModifiedTime.compareTo(otherLayerEntryTemplate.lastModifiedTime);
-      if (lastModifiedTimeComparison != 0) {
-        return lastModifiedTimeComparison;
+      int sourceLastModifiedTimeComparison =
```

Comment:
The order can be arbitrary, as the purpose is to be able to determine if two layers (list of `LayerEntries`) are "same" or not (in the sense whether they would lead to the same layer tarball.)

*However*, unrelated, now I think more about this, I think adding source file timestamps here as another criterion for checking equality may have a significant negative performance impact. That is, whenever a file is touched but not actually modified (e.g., copied the same resource file into the build directory), the cache will regard it as a different file. But the source file timestamps are only meaningful for `KEEP_ORIGINAL`, so for the default behavior and for setting a fixed timestamp over all files, I think this will result in a huge overhead as the cache will create a new tarball only to find out later that the layer already exists.

I will have to think about this a bit more. This could be tricky to optimize.

Reply:
So `target` is the only one that matters?

Reply:
No, source matters too, because for `KEEP_ORIGINAL`, the files of the cached layer (a tarball) are set with the timestamps of the source files in that case. And if the files in the layer should have different timestamps (e.g., when the user changes the configuration to `EPOCH_PLUS_SECOND` or sets a fixed timestamp, or the source timestamp changed while using `KEEP_ORIGINAL`), Jib does need to create a new cached layer. So, for these scenarios, the source file timestamps should be factored into determining whether the layers are same or different, and this is correct.

However, suppose the user never uses `KEEP_ORIGINAL`. Then we never look into source file timestamps, hence no need to factor them into the difference computation at all. However, this PR always take source timestamps into account. Therefore, for example, whenever the timestamp of a `.class` file is touched but the content remains exactly same, Jib will always create a new tarball first only to find out it did not have to. In this scenario where `KEEP_ORIGINAL` is never used, source timestamps do not play any role in creating a layer tarball. Did I make it clear?

Reply:
> I think adding source file timestamps here as another criterion for checking equality may have a significant negative performance impact.

The cache was already doing this. I was briefly confused. I'm restoring the `LayerEntriesSelector` changes (https://github.com/GoogleContainerTools/jib/pull/1818#issuecomment-515184225).